core/server/services/members/content-gating.js

```diff
@@ -23,7 +23,7 @@ function checkPostAccess(post, member) {
         return PERMIT_ACCESS;
     }
 
-    if (post.visibility === 'paid' && member.status === 'paid') {
+    if (post.visibility === 'paid' && (member.status === 'paid' || member.status === 'comped' || member.comped)) {
```

Comment:
@allouis, @rishabhgrg this isn't working. If I `console.log(member)` at this point the comped member always has `.status = 'free'` and no `comped` property .

Reply:
The problem was in test fixtures. There were missing initialization steps for member's subscription/customer relations. We'll need to make these fixtures more straight forward to use. Right now it's a labyrinth to navigate when adding a single new record to the fixtures :grimacing:

Reply:
Nice catch, minor thing that we should update the check here to use `!== free` instead of checking both comp and paid like rest of the codebase

Reply:
IMO, current approach - using "allowlist" of statuses is a better from security standpoint. It's much better to have bugs like current one where we don't expose sensitive data instead of accidentally exposing it to unwanted actors. 

If we are concerned with the logic consistency in other places that apply similar paid/free/comped filtering, like [email recipient list here](https://github.com/TryGhost/Ghost/blob/e87ded8d7206a2ca3ed3adf989b606df1e380c4d/core/server/services/mega/mega.js#L115-L117), I'd propose changing logic to one like in current PR, e.g: `status:['paid','comped']` ([NQL does support](https://github.com/NexesJS/NQL-Lang/blob/85ab577fe0cf7ff2bfaf49d24137d5222287655a/test/parser.test.js#L82-L90) `IN` style filtering).